### PR TITLE
MULE-19205: Migrate validation in ScatterGatherRouter#initialize

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: schedulers-tests/src/test/java/org/mule/test/integration/schedule/SchedulerConfigurationFailuresLazyInitTestCase.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import org.junit.rules.ExpectedException;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import io.qameta.allure.Feature;
+import io.qameta.allure.Features;
+import io.qameta.allure.Story;
+
+@Features({@Feature(LAZY_INITIALIZATION), @Feature(SCHEDULER)})
+@Story("Validations")
```
Comment (fsgonz) [inline schedulers-tests/src/test/java/org/mule/test/integration/schedule/SchedulerConfigurationFailuresLazyInitTestCase.java:35]: Use constants for the stories.

File: schedulers-tests/src/test/java/org/mule/test/integration/schedule/SchedulerConfigurationFailuresTestCase.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+
+@Feature(SCHEDULER)
+@Story("Validations")
```
Comment (fsgonz) [inline schedulers-tests/src/test/java/org/mule/test/integration/schedule/SchedulerConfigurationFailuresTestCase.java:29]: Use constants for the stories.

File: schedulers-tests/src/test/resources/org/mule/test/integration/invalid/core-scheduler-no-scheduling-strategy.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns="http://www.mulesoft.org/schema/mule/core"
+      xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd">
+
+    <flow name="scheduledFlow">
+        <scheduler>
+<!--             <scheduling-strategy> -->
```
Comment (fsgonz) [inline schedulers-tests/src/test/resources/org/mule/test/integration/invalid/core-scheduler-no-scheduling-strategy.xml:8]: Add an explanation  to indicate while these lines are commented if appropriate.

File: schedulers-tests/src/test/resources/org/mule/test/integration/invalid/pet-store-scheduler-no-scheduling-strategy.xml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:petstore="http://www.mulesoft.org/schema/mule/petstore"
+      xmlns="http://www.mulesoft.org/schema/mule/core"
+      xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+               http://www.mulesoft.org/schema/mule/petstore http://www.mulesoft.org/schema/mule/petstore/current/mule-petstore.xsd">
+
+    <flow name="scheduledFlow">
+        <petstore:pet-adoption-source>
+<!--             <scheduling-strategy> -->
```
Comment (fsgonz) [inline schedulers-tests/src/test/resources/org/mule/test/integration/invalid/pet-store-scheduler-no-scheduling-strategy.xml:10]: Add an explanation  to indicate while these lines are commented if appropriate.

**Review by fsgonz — CHANGES_REQUESTED** [review]

File: integration/src/test/java/org/mule/test/routing/RoutesLazyInitTestCase.java [hunk 5]
```diff
@@ -29,7 +30,7 @@
 import io.qameta.allure.Story;
 
 @Feature(ROUTERS)
-@Story(ROUND_ROBIN)
```
[the remark this answers is not in the document]
Comment (elrodro83) [reply integration/src/test/java/org/mule/test/routing/RoutesLazyInitTestCase.java:32]: it was wrong, this tests was using `SCATTER_GATHER`

**Review by fsgonz — APPROVED** [review]

**Pull request merged**